cosmic-core/api/src/main/java/org/apache/cloudstack/api/command/user/vpc/ListStaticRoutesCmd.java

```diff
@@ -90,12 +95,20 @@ public String getCommandName() {
         return s_name;
     }
 
-    private Optional<String> retrieveGateway() {
+    private Optional<String> retrieveGatewayCidr() {
         final Optional<String> gatewayCidr = Optional.of("0.0.0.0/0");
         if (gatewayId != null) {
             final VpcGateway gateway = _vpcService.getVpcPrivateGateway(gatewayId);
             gatewayCidr.of(NetUtils.ipAndNetMaskToCidr(gateway.getGateway(), gateway.getNetmask()));
         }
         return gatewayCidr;
     }
+
+    private Long retrieveVpcId() {
+        if (gatewayId != null) {
+            final VpcGateway gateway = _vpcService.getVpcPrivateGateway(gatewayId);
+            return gateway.getVpcId();
```

Comment:
`gateway` can be null here when a non existing private gateway id has been used. Please review @remibergsma